### PR TITLE
Distributed execution support of Hive with vineyard

### PR DESCRIPTION
**Pull request by vegetableysm**
What do these changes do?
-------------------------

- Support distributed execution for hive with vineyard.
- Build the docker cluster with scripts.
- The hdfs related stuffs have been removed.

-------------------------
Currently, the vineyard jar cannot run directly on hive because of dependency conflicts. You can run it  temporarily by reverting to an older version of guava dependent by vineyard.
This problem will be fixed in #1682

### CODE REVIEW

**Comment by vegetableysm** [comment]
I try to add location info for splits and Tez is able to get the information I provided, but failed to request a node resource.

Here is the log:
tez

We can see that yarn task scheduler service get the node hostname of "hadoop-yarn-nm-0" but failed to assign container. When the tez client asks for "/default-rack," it matches the hadoop-yarn-nm-0 node ("not match" in the figure should be " match". It is a mistake). Since the current three nodes are at /default-rack, if the requested resource is "/default-rack", which node the task is assigned to is random. There is no guarantee that the task will be assigned to the node with the corresponding data.

I have no idea why the allocation failed.

File: java/core/src/main/java/io/v6d/core/client/Context.java [hunk 1]
```diff
@@ -34,6 +36,13 @@ public static synchronized IPCClient getClient() throws VineyardException {
         return client;
     }
 
+    public static synchronized InstanceID getClientInstanceId() {
```
Comment (sighingnow) [inline java/core/src/main/java/io/v6d/core/client/Context.java:39]: getClientInstanceId -> getInstanceId

File: java/modules/basic/src/main/java/io/v6d/modules/basic/arrow/TableBuilder.java [hunk 2]
```diff
@@ -87,4 +89,43 @@ public ObjectMeta seal(Client client) throws VineyardException {
 
         return client.createMetaData(meta);
     }
+
+    public static ObjectMeta fromRecordMeta(Client client, ObjectMeta schemaMeta, List<ObjectMeta> recordBatchMetas, int columns) throws VineyardException {
```
Comment (sighingnow) [inline java/modules/basic/src/main/java/io/v6d/modules/basic/arrow/TableBuilder.java:93]: fromRecordMeta -> fromRecordBatchMeta?

We don't have the concept of "record".

File: java/hive/src/main/java/io/v6d/hive/ql/io/VineyardInputFormat.java [hunk 3]
```diff
@@ -156,63 +178,50 @@ class VineyardRecordReader implements RecordReader<NullWritable, RecordWrapperWr
...
+        for (int i = (int)split.getStart(); i < split.getLength() + split.getStart(); i++) {
+            val batch = table.getBatch(i);
+            batch.setResolver(resolver);
+            this.batches[this.recordBatchIndex++] = batch;
         }
+        schema = table.getSchema().getSchema();
         // reset to the beginning
         this.recordBatchIndex = -1;
         this.recordBatchInnerIndex = 0;
+        in.close();
```
Comment (sighingnow) [inline java/hive/src/main/java/io/v6d/hive/ql/io/VineyardInputFormat.java:224]: Use try-with-resource style for file I/O to simplify the logic for open/close.

File: java/core/src/main/java/io/v6d/core/client/Context.java (hunk 1 above)
Comment (vegetableysm) [reply java/core/src/main/java/io/v6d/core/client/Context.java]: Done

File: java/modules/basic/src/main/java/io/v6d/modules/basic/arrow/TableBuilder.java (hunk 2 above)
Comment (vegetableysm) [reply java/modules/basic/src/main/java/io/v6d/modules/basic/arrow/TableBuilder.java]: OK

File: java/hive/src/main/java/io/v6d/hive/ql/io/VineyardInputFormat.java (hunk 3 above)
Comment (vegetableysm) [reply java/hive/src/main/java/io/v6d/hive/ql/io/VineyardInputFormat.java]: Done

**Review by sighingnow — COMMENTED** [review]
Remove hdfs related stuffs in the docker-compose configuration files.

**Comment by sighingnow** [comment]
Rebase to main please.

**Comment by vegetableysm** [comment]
> Rebase to main please.

Done

**Review by sighingnow — COMMENTED** [review]
One more comment: can we unify the dockerfiles under `distributed/` to existing `docker/` stuffs?

**Review by sighingnow — APPROVED** [review]

**Pull request merged**